OpenRA.Game/World.cs

```diff
@@ -207,30 +207,13 @@ internal World(ModData modData, Map map, OrderManager orderManager, WorldType ty
 			Selection = WorldActor.Trait<ISelection>();
 			OrderValidators = WorldActor.TraitsImplementing<IValidateOrder>().ToArray();
 
-			// Reset mask
 			LongBitSet<PlayerBitMask>.Reset();
 
-			// Add players
 			// Create an isolated RNG to simplify synchronization between client and server player faction/spawn assignments
 			var playerRandom = new MersenneTwister(orderManager.LobbyInfo.GlobalSettings.RandomSeed);
 			foreach (var cmp in WorldActor.TraitsImplementing<ICreatePlayers>())
 				cmp.CreatePlayers(this, playerRandom);
 
-			// Set defaults for any unset stances
-			foreach (var p in Players)
-			{
-				if (!p.Spectating)
-					AllPlayersMask = AllPlayersMask.Union(p.PlayerMask);
-
-				foreach (var q in Players)
-				{
-					SetUpPlayerMask(p, q);
```

Comment:
We now appear to only be setting up the masks in `CreateMPPlayers`; can you explain why this doesn't break the campaigns? (allies01 doesn't show any obvious regressions against the enemy and ally players)?

Reply:
`CreateMPPlayers` is not removed from the campaign rules, from what I know, so that's why.

Reply:
Ah, I see - and this trait already handles other important things that will break campaign maps if it were removed. We really should think about renaming that trait in a followup PR.